ironic/drivers/modules/drac/management.py

```diff
@@ -302,14 +314,170 @@ def set_boot_device(node, device, persistent=False):
+    def clear_job_queue(self, task):
+        """Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: RedfishError on an error.
+        """
+        system = redfish_utils.get_system(task.node)
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                            "'sushy-oem-idrac' failed for node %(node)s. "
+                            "Ensure it is installed. Error: %(error)s") %
+                            {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                delete_job_response = oem_manager.delete_jobs(task,
+                                                job_ids=['JID_CLEARALL'])
+            except sushy.exceptions.SushyError as e:
+                error_msg = ("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to clear job queue"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+
+        if delete_job_response.status_code == _JOB_RESPONSE_CODE:
+            LOG.info("Job queue cleared for node %(node)s via OEM" %
+                    {'node': task.node.uuid})
+        else:
+            error_msg = ("Failed to clear job queue, node : %(node)s " %
+                        {'node': task.node.uuid})
+            LOG.error(error_msg)
+            raise exception.RedfishError(error=error_msg)
+
+        return None
+
+    @METRICS.timer('DracRedfishManagement.reset_idrac')
+    @base.clean_step(priority=0)
+    def reset_idrac(self, task, ready_wait_time=30):
+        """Reset the iDRAC.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: RedfishError on an error.
+        """
+        system = redfish_utils.get_system(task.node)
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                           "'sushy-oem-idrac' failed for node %(node)s. "
+                           "Ensure it is installed. Error: %(error)s") %
+                         {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                reset_job_response = oem_manager.reset_idrac(task)
```

Comment:
The above line does a graceful reset.  We need to do a force reset.

Reply:
Done, Added force='Force' parameter to reset_idrac method.